### PR TITLE
Remove `git add` in precommit

### PR DESCRIPTION
**Pull request by jridgewell**
It frequently fails for no reason (blocking commits!), and even when it does work it breaks incremental commits.

| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Deprecations?            | No
| Spec Compliancy?         | No
| Tests Added/Pass?        | No
| Fixed Tickets            |
| License                  | MIT
| Doc PR                   |
| Dependency Changes       |

### CODE REVIEW

**Comment by sarupbanskota** [comment]
ahh so it wasn't me. struggled for hours with this on https://github.com/babel/babel/pull/5911

File: package.json [hunk 1]
```diff
@@ -65,8 +65,7 @@
   },
   "lint-staged": {
     "*.js": [
-      "eslint --format=codeframe --fix",
-      "git add"
+      "eslint --format=codeframe --fix"
```
Comment (sarupbanskota) [inline package.json]: Might also be worth introducing something for prettier? https://github.com/prettier/prettier#pre-commit-hook

Comment (existentialism) [inline package.json]: Prettier is already ran via `eslint-plugin-prettier`.

**Comment by existentialism** [comment]
Fine with removing (though it works fine for me?), but if we do... then we should drop it from the other projects too?

File: package.json (hunk 1 above)
Comment (sarupbanskota) [inline package.json]: ah gotcha. somehow it seemed like it didn't run for me locally, but i'll check it out

**Comment by Andarist** [comment]
Weird, Im using this in multiple projects with `git add` without any problems. I would suggest finding out why it fails sometimes, as this is allows automation of pretty-printing and saves quite some time. 

Without it I can even imagine it will **break** commits if staged file gets reformatted, it will not get committed at all, which is probably something noone really wants and can lead to a lot of confusion.

**Comment by sarupbanskota** [comment]
When it breaks for me, it typically goes like this

```
--- gig/babel ‹fix/plugin-ordering-failing-tests* AMDM› » git commit -m "Merge 7.0 + update tests + improve formatting"

> husky - npm run -s precommit
> husky - node v8.0.0

 ❯ Running tasks for *.js
   ✔ eslint --format=codeframe --fix
   ✖ git add
     → remove the file manually to continue.
🚫 git add found some errors. Please fix them and try committing again.

fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.
fatal: Unable to create '/Users/sarupbanskota/gig/babel/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.


> husky - pre-commit hook failed (add --no-verify to bypass)
> husky - to debug, use 'npm run precommit'
```

**Comment by jridgewell** [comment]
We can remove the fix portion and just have it fail if linting fails.

Mine always fails with ".git/index.lock.lock: File exists"

**Comment by benjamn** [comment]
I support this change. The current behavior makes `git commit -p` useless, which is an unpleasant surprise for anyone who knows how to use `git`!

**Pull request closed without merging**